app/controllers/event_log_files_controller.rb

```diff
@@ -132,16 +140,34 @@ def date_to_time(date)
   end
 
   # helper method to dynamically generate the valid event log file event types
-  def get_event_types
+  def get_event_types(fields)
     pick_list_values = []
-    fields = @client.describe_sobject("EventLogFile")["fields"]
     for field in fields
       if field["name"] == "EventType"
-         field["picklistValues"].each {|v| pick_list_values.push(v["value"])}
+        field["picklistValues"].each {|v| pick_list_values.push(v["value"])}
         break
       end
     end
     return pick_list_values.dup.unshift(ALL_EVENTS_TYPE)
   end
 
+  # helper method to identify whether an org has 1-hr ELF
+  def one_hour_elf?(fields)
+    for field in fields
+      if field["name"] == "Interval" or field["name"] == "Sequence"
+        return "1"
+      end
+    end
+    return nil
+  end
+
+  def metadata_cached?
+    session[:event_types] || session[:has_one_hr_elf]
```

Comment:
I'd check if both session variables are set. If either one is not set, reload the cache.

`session[:event_types] && session[:has_one_hr_elf]`